src/logic/safe/transactions/send.js

```diff
@@ -37,24 +39,49 @@ export const approveTransaction = async (
       from: sender,
     },
   )
-  const receipt = await safeInstance.approveHash(contractTxHash, { from: sender })
 
-  await saveTxToHistory(
-    safeInstance,
-    to,
-    valueInWei,
-    data,
-    operation,
-    nonce,
-    receipt.tx, // tx hash,
-    sender,
-    TX_TYPE_CONFIRMATION,
-  )
+  try {
+    const web3 = getWeb3()
+    const contract = new web3.eth.Contract(GnosisSafeSol.abi, safeInstance.address)
+
+    const transactionHash = await contract.methods.approveHash(txHash)
+      .send({
+        from: sender,
+      }).once('transactionHash', () => {
```

Comment:
Let's keep the logic function simple and move this even emitter logic to the action. So we would just do
```
const transactionHash = contract.methods.approveHash(txHash)

return transactionHash
```
This way we won't have to deal with wrapping notifications, posting tx to history (possibly later) or anything else with the dispatch